src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -472,7 +477,9 @@ export class MockHttpSocket extends MockSocket {
           // flush the write buffer to trigger the callbacks.
           // This way, if the request stream ends in the write callback,
           // it will indeed end correctly.
-          this.flushWriteBuffer()
+          if (!this.isPassthrough) {
+            this.flushWriteBuffer()
```

Comment:
As I've mentioned, this introduces an impossible state. You won't know if the request is passthrough until the handler is finished running. For the handler to finish running, it must be able to read the request's body, for which we must flush these first chunks.